plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergUtil.java

```diff
@@ -669,7 +671,27 @@ public static Map<String, String> createTableProperties(ConnectorTableMetadata t
         if (tableMetadata.getComment().isPresent()) {
             propertiesBuilder.put(TABLE_COMMENT, tableMetadata.getComment().get());
```

Comment:
I think we could currently set the table comment through `extra_properties`.

```
assertQueryFails(
                "CREATE TABLE create_table_with_illegal_extra_properties (c1 integer) WITH (extra_properties = MAP(ARRAY['comment'], ARRAY['table comment']))",
                ".*Illegal keys in extra_properties: \\[comment\\].*");
```

The statement above doesn't fail (even though it would be expected to fail).

Do add **ALL** the properties which can be set by Trino in a static map and check for intersection with `extra_properties`.

Reply:
cc @Praveen2112  we probably have the same issue with Hive's `extra_properties`

Reply:
The comment key is only added when there are comments.
https://github.com/trinodb/trino/blob/6ef1a1fd6da6c35a6cf719da29e456c2559d9ffb/plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergUtil.java#L671-L673

Perhaps it should be defined in SUPPORTED_PROPERTIES.
https://github.com/trinodb/trino/blob/6ef1a1fd6da6c35a6cf719da29e456c2559d9ffb/plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergTableProperties.java#L51-L59

Currently, the comment key constant is imported from HiveMetadata, and I believe this is incorrect.
https://github.com/trinodb/trino/blob/6ef1a1fd6da6c35a6cf719da29e456c2559d9ffb/plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergUtil.java#L100

What do you think about creating the TABLE_COMMENT constant and adding it to the SUPPORTED_PROPERTIES?

Reply:
No. Not a valid idea.

The comment can be set through `CREATE TABLE xxx (....) COMMENT '....'` syntax.

As mentioned already 

> Do add ALL the Iceberg native table properties which can be set by Trino in a static map and check for intersection with extra_properties

Reply:
It would be nice to disallow setting table comment via extra_properties. Please fix and add a test.

Reply:
@findinpath @ebyhr Done! I made it in another commit to make the review easier, and I will squash the commits once it's approved.